src/Contexts/filter-controller.js

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useReducer } from 'react'
```

Comment:
Suggestion: I think the file naming could be better, Like FilterProvider or FilterContext

Reply:
done